quixstreams/dataframe/series.py

```diff
@@ -9,11 +9,34 @@
 from quixstreams.core.stream.stream import Stream
 from quixstreams.models.messagecontext import MessageContext
 from .base import BaseStreaming
-from .exceptions import InvalidOperation
+from .exceptions import InvalidOperation, MissingColumn, InvalidColumnReference
 
 __all__ = ("StreamingSeries",)
 
 
+def _getitem(d: Mapping, column_name: Union[str, int]) -> object:
+    """
+    Special error handling around column referencing with SDF.
+
+    :param d: a dict-like object (usually just a dict).
+    :param column_name: the column name.
+
+    :return: Nested data from column name
+    """
+    try:
+        return d[column_name]
+    except KeyError:
+        raise MissingColumn(
+            f"Column name '{column_name}' does not exist in the StreamingDataFrame"
+        )
```

Comment:
Let's  rename "MissingColumn" to "ColumnDoesNotExist" (as it was in the task).
On top of it, the text says "does not exist" anyway :)